### PR TITLE
Credentials connector code cleanup

### PR DESCRIPTION
**Pull request by revit13**
The credentials connector is used to return the dataset credentials in order for the manager to register them in Vault such that the modules could later retrieve them from Vault tand use them o access the dataset.
As the dataset credentials are now retrieved using Vault plugin directly by the modules as described in issue #406; credentials connector code becomes absolute.
This PR which addresses issue #547 is about cleaning all related credentials connector code.

### CODE REVIEW

File: charts/m4d/templates/m4d-config.yaml [hunk 1]
```diff
@@ -8,14 +8,10 @@ data:
...
-  CREDENTIALS_PROVIDER_NAME: {{ .Values.coordinator.credentialsManager | quote }}
-  CREDENTIALS_CONNECTOR_URL: {{ .Values.coordinator.credentialsManagerConnectorURL | default (printf "%s-connector:80" .Values.coordinator.catalog) | quote }}
   MAIN_POLICY_MANAGER_NAME: {{ .Values.coordinator.policyManager | quote }}
   MAIN_POLICY_MANAGER_CONNECTOR_URL: {{ .Values.coordinator.policyManagerConnectorURL | default (printf "%s-connector:80" .Values.coordinator.policyManager) | quote }}
   USE_EXTENSIONPOLICY_MANAGER: "false" # deprecated
   VAULT_ADDRESS: {{ .Values.coordinator.vault.address | quote }}
-  VAULT_DATASET_HOME: {{ printf "m4d/dataset-creds/" | quote }} # temporary
-  VAULT_DATASET_MOUNT: {{ printf "/v1/sys/mounts/m4d/dataset-creds" | quote }} # temporary
   USER_VAULT_ADDRESS: {{ .Values.coordinator.vault.address | quote }} # deprecated
   USER_VAULT_PATH: "external" # deprecated
```
Comment (shlomitk1) [inline charts/m4d/templates/m4d-config.yaml:16]: USER_xxx should not be a part of m4d-config. Are they needed at all?

File: manager/main.go [hunk 2]
```diff
@@ -182,15 +182,13 @@ func main() {
 	}
 }
 
-// init vault client and mount the base directory for storing credentials
+// init vault client
+// TODO: the vault client should be used to add roles for the modules.
 func initVaultConnection() (vault.Interface, error) {
```
Comment (shlomitk1) [inline manager/main.go:187]: In this implementation manager does not communicate with vault at all. This means:
1. No need for VAULT_TOKEN passed to the manager pod
2. No need for vault.Interface inside manager

**Comment by froesef** [comment]
There are going to be conflicts with #541 . So we should coordinate the merge order of the PRs.

File: manager/main.go (hunk 2 above)
Comment (revit13) [reply manager/main.go:187]: I left it as is as the next step would be to let the manager to set Vault policies and Vault roles for modules.

File: charts/m4d/templates/m4d-config.yaml (hunk 1 above)
Comment (revit13) [reply charts/m4d/templates/m4d-config.yaml:16]: These should have been removed. I'll create a separate PR for it. Thanks

**Review by roee88 — APPROVED** [review]

**Review by shlomitk1 — APPROVED** [review]

**Review by froesef — APPROVED** [review]

**Review by rohithdv — APPROVED** [review]
Once this PR is merged I will raise a PR for the wkc connector changes. Thanks.

**Pull request merged**